pilot/pkg/xds/debug.go

```diff
@@ -580,6 +583,60 @@ func (s *DiscoveryServer) adsz(w http.ResponseWriter, req *http.Request) {
 	writeJSON(w, adsClients)
 }
 
+// ecdsz implements a status and debug interface for ECDS.
+// It is mapped to /debug/ecdsz
+func (s *DiscoveryServer) ecdsz(w http.ResponseWriter, req *http.Request) {
+	if s.handlePushRequest(w, req) {
+		return
+	}
+	proxyID, con := s.getDebugConnection(req)
+	if con == nil {
+		s.errorHandler(w, proxyID, con)
+		return
+	}
+
+	if s.Generators[v3.ExtensionConfigurationType] != nil {
```

Comment:
maybe we should just make config_dump able to take in a list of generator keys. We can have shortcuts sa well. So something like `config_dump?types=ecds`.

Then we do not need a bunch of new methods for every type we add?

Reply:
right now, `adsz` have something like what your said, `confug_dump` will dump everything .

https://github.com/istio/istio/blob/e3904269b7bd95207fdccd2be6f14276a3ad470a/pilot/pkg/xds/debug.go#L177

https://github.com/istio/istio/blob/e3904269b7bd95207fdccd2be6f14276a3ad470a/pilot/pkg/xds/debug.go#L195

I'm not sure that `config_dump?types=ecds` is a good idea or not.

Reply:
tracking in https://github.com/istio/istio/issues/37504